controllers/autoscaling_test.go

```diff
@@ -0,0 +1,136 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	github2 "github.com/google/go-github/v31/github"
+	"github.com/summerwind/actions-runner-controller/api/v1alpha1"
+	"github.com/summerwind/actions-runner-controller/github"
+	"k8s.io/klog/klogr"
+	"testing"
+)
+
+func TestDetermineDesiredReplicas_RepositoryRunner(t *testing.T) {
+	intPtr := func(v int) *int {
+		return &v
+	}
+
+	testcases := []struct {
+		fixed               *int
+		max                 *int
+		min                 *int
+		workflowRunStatuses []string
+		want                int
+	}{
+		// 3 demanded, max at 3
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "in_progress", "in_progress", "completed"},
+			want:                3,
+		},
+		// 3 demanded, max at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(2),
+			workflowRunStatuses: []string{"queued", "in_progress", "in_progress", "completed"},
+			want:                2,
+		},
+		// 2 demanded, min at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "in_progress", "completed"},
+			want:                2,
+		},
+		// 1 demanded, min at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "completed"},
+			want:                2,
+		},
+		// 1 demanded, min at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"in_progress", "completed"},
+			want:                2,
+		},
+		// 1 demanded, min at 1
+		{
+			min:                 intPtr(1),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "completed"},
+			want:                1,
+		},
+		// 1 demanded, min at 1
+		{
+			min:                 intPtr(1),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"in_progress", "completed"},
+			want:                1,
+		},
+		// fixed at 3
+		{
+			fixed: intPtr(3),
+			want:  3,
+		},
+	}
+
+	for i := range testcases {
+		tc := testcases[i]
+
+		log := klogr.New()
+		r := &RunnerDeploymentReconciler{
+			Log: log,
+			GitHubClient: &github.Client{
+				ListRepositoryWorkflowRuns: func(ctx context.Context, owner, repo string, opts *github2.ListOptions) (runs *github2.WorkflowRuns, response *github2.Response, err error) {
```

Comment:
I'm sure what this code is doing. However, we use the fake server for tests that need to work with the GitHub API. Is it possible to modify this test to be dependent on the fake server?

Reply:
Great suggestion 👍

Addressed in 4ebc4e8. Please note that I've enhanced the fake server creation to customize the API response per test run. Hope you like it.